### PR TITLE
refactor swipe up/down 

### PR DESCRIPTION
**Pull request by nalinbhardwaj**
tested on iOS

closes https://github.com/daimo-eth/daimo/issues/717

### CODE REVIEW

File: apps/daimo-mobile/src/view/screen/HistoryOpScreen.tsx [hunk 1]
```diff
@@ -42,22 +43,31 @@ import {
...
 
-type Props = { account: Account; op: DisplayOpEvent };
+type Props = NativeStackScreenProps<
+  ParamListBottomSheet,
+  "BottomSheetHistoryOp"
+>;
 
-export const ToggleBottomSheetContext = createContext((expand: boolean) => {});
+export const ChangeBottomSheetSnapsContext = createContext(
+  (snaps: 2 | 3) => {}
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/HistoryOpScreen.tsx:52]: what does this do?

File: apps/daimo-mobile/src/view/screen/HistoryOpScreen.tsx [hunk 2]
```diff
@@ -66,12 +76,17 @@ function HistoryOpScreenInner({ account, op }: Props) {
...
+  const nav = useNav();
+
   return (
     <View style={ss.container.screen}>
       <ScreenHeader
         title="Transfer"
         onExit={() => {
-          toggleBottomSheet(false); // Collapse to small height
+          if (nav.canGoBack()) {
+            changeBottomSheetSnaps(3);
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/HistoryOpScreen.tsx:87]: it feels weird for the HistoryOpScreen (inner component) to have to change the SwipeUpDown (outer component) this way

if i understand correctly:
- SwipeUpDown always has 2 snap points (~25% open and 100% open)
- in some situations, it has a third (half screen)
- `changeBottomSheetSnaps` is how we're toggling this

**Review by dcposch — APPROVED** [review]
LGTM. Left comments.

I did prefer the swipe-down gesture to close the Transfer page, but bug-free is more important.

File: apps/daimo-mobile/src/view/screen/HistoryOpScreen.tsx (hunk 1 above)
Comment (nalinbhardwaj) [reply apps/daimo-mobile/src/view/screen/HistoryOpScreen.tsx:52]: added a comment:
> Allows the HistoryOpScreen to change the bottom sheet snap point count.
> This allows the bottom sheet to be dismissed when the user exits the detail
> screen and only display the half screen snap point when the user is on the
> detail screen.

File: apps/daimo-mobile/src/view/screen/HistoryOpScreen.tsx (hunk 2 above)
Comment (nalinbhardwaj) [reply apps/daimo-mobile/src/view/screen/HistoryOpScreen.tsx:87]: yep that's correct -- when on the list screen we want to show three snap points with the lowest being the default dismissed version, when on the detail op screen we want to show two snap points with the lower being the middle "active" (with changed background) version

Comment (nalinbhardwaj) [reply apps/daimo-mobile/src/view/screen/HistoryOpScreen.tsx:87]: not super happy with the ugliness either but best I could come up with.

**Pull request merged**